From caf064c1349ab6c053fcf288625e27a9a72e7439 Mon Sep 17 00:00:00 2001
From: Alex Caithness <acaithness@cclgroupltd.com>
Date: Fri, 1 Jul 2022 18:45:22 +0100
Subject: [PATCH 02/19] Fixed broken report icons/layout Added `category` to
 PluginSpec `crunch_artifacts` now in charge of creating the category output
 folder (could be done elsewhere if more appropriate)

---
 aleapp.py                              |  6 +++++-
 aleappGUI.py                           | 15 ++-------------
 plugin_loader.py                       |  5 +++--
 scripts/artifacts/calllogs.py          |  7 ++++---
 scripts/artifacts/contacts.py          |  5 ++++-
 scripts/artifacts/usagestatsVersion.py |  5 ++++-
 6 files changed, 22 insertions(+), 21 deletions(-)

diff --git a/aleapp.py b/aleapp.py
index 43acaee..0ffa9a5 100644
--- a/aleapp.py
+++ b/aleapp.py
@@ -1,6 +1,7 @@
 import argparse
 import io
 import os
+import os.path
 import typing
 
 import plugin_loader
@@ -153,7 +154,10 @@ def crunch_artifacts(
         if files_found:
             logfunc()
             #process_artifact(files_found, key, artifact_pretty_name, seeker, out_params.report_folder_base, wrap_text)
-            plugin.method(files_found, out_params.report_folder_base, seeker, wrap_text)
+            category_folder = os.path.join(out_params.report_folder_base, plugin.category)
+            if not os.path.exists(category_folder):
+                os.mkdir(category_folder)
+            plugin.method(files_found, category_folder, seeker, wrap_text)
             for pathh in files_found:
                 if pathh.startswith('\\\\?\\'):
                     pathh = pathh[4:]
diff --git a/aleappGUI.py b/aleappGUI.py
index 1d7afb1..545b775 100644
--- a/aleappGUI.py
+++ b/aleappGUI.py
@@ -1,16 +1,11 @@
 import typing
-
 import aleapp
-import os
 import PySimpleGUI as sg
-import sys
 import webbrowser
-
 import plugin_loader
 from scripts.ilapfuncs import *
 from scripts.version_info import aleapp_version
 from time import process_time, gmtime, strftime
-#from scripts.ilap_artifacts import *
 from scripts.search_files import *
 
 MODULE_START_INDEX = 1000
@@ -55,6 +50,7 @@ def ValidateInput(values, window):
 
     return True, ext_type
 
+
 # initialize CheckBox control with module name   
 def CheckList(mtxt, lkey, mdstring, disable=False):
     if mdstring == 'test1' or mdstring == 'test2' : #items in the if are modules that take a long time to run. Deselects them by default.
@@ -71,17 +67,10 @@ def pickModules():
 
     loader = plugin_loader.PluginLoader()
 
-    #script_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'scripts', 'artifacts')
-
-    # Create sorted dict from 'tosearch' dictionary based on plugin category
-    #sorted_tosearch = {k: v for k, v in sorted(tosearch.items(), key=lambda item: item[1][0].upper())}
-    # sorted_tosearch = {p.name: p for p in loader.plugins}
-
     indx = MODULE_START_INDEX     # arbitrary number to not interfere with other controls
     for plugin in sorted(loader.plugins, key=lambda p: p.name.upper()):
-        #disabled = False if plugin_name != 'usagestatsVersion' else True # usagestatsVersion is REQUIRED
         disabled = plugin.module_name == 'usagestatsVersion'
-        mlist.append(CheckList(f'{plugin.name} [{plugin.module_name}]', indx, plugin.name, disabled))
+        mlist.append(CheckList(f'{plugin.category} [{plugin.name} - {plugin.module_name}.py]', indx, plugin.name, disabled))
         indx = indx + 1
 
 
diff --git a/plugin_loader.py b/plugin_loader.py
index 3636f03..94f6635 100644
--- a/plugin_loader.py
+++ b/plugin_loader.py
@@ -10,6 +10,7 @@ PLUGINPATH = pathlib.Path("./scripts/artifacts")
 class PluginSpec:
     name: str
     module_name: str
+    category: str
     search: str
     method: typing.Callable  # todo define callable signature
 
@@ -37,11 +38,11 @@ class PluginLoader:
             except AttributeError:
                 continue  # no artifacts defined in this plugin
 
-            for name, (search, func) in mod_artifacts.items():
+            for name, (category, search, func) in mod_artifacts.items():
                 #self._plugins.append(PluginSpec(name, search, func))
                 if name in self._plugins:
                     raise KeyError("Duplicate plugin")
-                self._plugins[name] = PluginSpec(name, py_file.stem, search, func)
+                self._plugins[name] = PluginSpec(name, py_file.stem, category, search, func)
 
     @property
     def plugins(self) -> typing.Iterable[PluginSpec]:
diff --git a/scripts/artifacts/calllogs.py b/scripts/artifacts/calllogs.py
index eda3d66..f07a2e8 100644
--- a/scripts/artifacts/calllogs.py
+++ b/scripts/artifacts/calllogs.py
@@ -81,7 +81,8 @@ def get_calllogs(files_found, report_folder, seeker, wrap_text):
 
 # 'calllogs':('Call Logs', ('**/com.android.providers.contacts/databases/contact*', '**/com.sec.android.provider.logsprovider/databases/logs.db*')),
 __artifacts__ = {
-    "Call Logs":
-        (('**/com.android.providers.contacts/databases/contact*', '**/com.sec.android.provider.logsprovider/databases/logs.db*'),
-         get_calllogs)
+    "Call Logs":(
+        "Call Logs",
+        ('**/com.android.providers.contacts/databases/contact*', '**/com.sec.android.provider.logsprovider/databases/logs.db*'),
+        get_calllogs)
 }
diff --git a/scripts/artifacts/contacts.py b/scripts/artifacts/contacts.py
index b55e4fb..e96ce68 100644
--- a/scripts/artifacts/contacts.py
+++ b/scripts/artifacts/contacts.py
@@ -79,5 +79,8 @@ def get_contacts(files_found, report_folder, seeker, wrap_text):
 
 # 'contacts':('Contacts', ('**/com.android.providers.contacts/databases/contact*', '**/com.sec.android.provider.logsprovider/databases/logs.db*')),
 __artifacts__ = {
-    "Contacts": (('**/com.android.providers.contacts/databases/contact*', '**/com.sec.android.provider.logsprovider/databases/logs.db*'), get_contacts)
+    "Contacts": (
+        "Contacts",
+        ('**/com.android.providers.contacts/databases/contact*', '**/com.sec.android.provider.logsprovider/databases/logs.db*'),
+        get_contacts)
 }
\ No newline at end of file
diff --git a/scripts/artifacts/usagestatsVersion.py b/scripts/artifacts/usagestatsVersion.py
index c074644..6f5db88 100644
--- a/scripts/artifacts/usagestatsVersion.py
+++ b/scripts/artifacts/usagestatsVersion.py
@@ -55,5 +55,8 @@ def get_usagestatsVersion(files_found, report_folder, seeker, wrap_text):
 
 
 __artifacts__ = {
-    "Usage Stats Version": (('*/system/usagestats/*/version', '*/system_ce/*/usagestats/version'), get_usagestatsVersion)
+    "Usage Stats Version": (
+        "Usage Stats",
+        ('*/system/usagestats/*/version', '*/system_ce/*/usagestats/version'),
+        get_usagestatsVersion)
 }
\ No newline at end of file
-- 
2.34.1

